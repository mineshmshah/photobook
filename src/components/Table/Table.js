import React from 'react';
import { useSelector } from 'react-redux';
import {
  TableBlock,
  TableContainer,
  TableData,
  TableHead,
  TableHeader,
  TableRow,
  TableBody,
} from './Table.styles';

export const Table = () => {
  const data = useSelector((state) => state.data.value);

  return (
    <TableBlock>
      <TableContainer>
        <TableHead>
          <TableRow>
            <TableHeader>First Name</TableHeader>
            <TableHeader>Last Name</TableHeader>
            <TableHeader>e-mail</TableHeader>
            <TableHeader>Date of Birth</TableHeader>
            <TableHeader>Industry</TableHeader>
            <TableHeader>Salary</TableHeader>
            <TableHeader>Years of Experience</TableHeader>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map((item) => (
            <TableRow key={item.id}>
              <TableData>{item.first_name}</TableData>
              <TableData>{item.last_name}</TableData>
              <TableData>{item.email}</TableData>
              <TableData>{item.date_of_birth}</TableData>
              <TableData>{item.industry}</TableData>
              <TableData>{item.salary}</TableData>
              <TableData>{item.years_of_experience}</TableData>
            </TableRow>
          ))}
        </TableBody>
      </TableContainer>
    </TableBlock>
  );
};

Table.propTypes = {};
