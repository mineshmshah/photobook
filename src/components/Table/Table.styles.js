import styled from 'styled-components/macro';

export const TableBlock = styled.div`
  max-width: 100%;
  padding: 16px;
`;

export const TableContainer = styled.table`
  font-family: Roboto, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
    Oxygen, Ubuntu, Cantarell, Open Sans, Helvetica Neue, sans-serif;
  width: 100%;
  border-collapse: collapse;
  border-radius: 4px 4px 0 0;
  overflow: hidden;
  text-align: left;
`;

export const TableRow = styled.tr`
  height: 50px;
`;

export const TableHeader = styled.th`
  font-weight: 400;
  font-size: 14px;
  color: white;
  padding: 8px 16px;
`;
export const TableData = styled.td`
  padding: 8px 16px;
  color: gray;
  font-size: 12px;
`;

export const TableHead = styled.thead`
  height: 8px;
  ${TableRow} {
    background-color: #484848;
  }
`;

export const TableBody = styled.tbody`
  ${TableRow} {
    background-color: #efefef;
  }
`;
