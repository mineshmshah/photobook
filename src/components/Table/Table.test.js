import React from 'react';
import { render } from '@testing-library/react';
import { Table } from './Table';

describe('Components/Table', () => {
  it('should render', () => {
    const { container } = render(<Table data={[]} />);
    expect(container).toMatchSnapshot();
  });
});
