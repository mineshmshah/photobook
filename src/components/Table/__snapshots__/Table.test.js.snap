// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Table should render 1`] = `
<div>
  <div
    class="Tablestyles__TableBlock-sc-1s0jmbg-0 YYiKP"
  >
    <table
      class="Tablestyles__TableContainer-sc-1s0jmbg-1 fyCJcI"
    >
      <thead
        class="Tablestyles__TableHead-sc-1s0jmbg-5 hKGmRx"
      >
        <tr
          class="Tablestyles__TableRow-sc-1s0jmbg-2 eXhRhL"
        >
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            First Name
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            Last Name
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            e-mail
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            Date of Birth
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            Industry
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            Salary
          </th>
          <th
            class="Tablestyles__TableHeader-sc-1s0jmbg-3 gPnLjD"
          >
            Years of Experience
          </th>
        </tr>
      </thead>
      <tbody
        class="Tablestyles__TableBody-sc-1s0jmbg-6"
      />
    </table>
  </div>
</div>
`;
